src/draive/scope/metrics.py

```diff
@@ -375,20 +376,32 @@ async def _summary(self) -> str:
 
         return summary
 
-    # TODO: combine all combinable metrics this way
-    # Warning: this method is not using lock
-    def _total_tokens_usage(self) -> TokenUsage:
-        total_usage: TokenUsage = TokenUsage()
-        match self._metrics.get(TokenUsage):
-            case TokenUsage() as usage:
-                total_usage.combine_metric(usage)
+    async def _combine_metric(self, metric: type[CombinableScopeMetric]) -> CombinableScopeMetric:
+        combined_metric = metric()
+        match self._metrics.get(metric):
+            case metric() as usage:
+                combined_metric.combine_metric(usage)
             case _:
                 pass
 
         for child in self._nested_traces:
-            total_usage.combine_metric(child._total_tokens_usage())
+            combined_metric.combine_metric(await child._combine_metric(metric))
+
+        return combined_metric
 
-        return total_usage
+    async def _combine_metrics(
+            self,
+            metrics: None | Iterable[type[CombinableScopeMetric]] = None
```

Comment:
we don't need ability to combine selected metrics, we should focus on combining all